src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/attributes/edit.svelte

```diff
@@ -0,0 +1,112 @@
+<script lang="ts">
+    import { goto, invalidate } from '$app/navigation';
+    import { base } from '$app/paths';
+    import { page } from '$app/stores';
+    import { Submit, trackError, trackEvent } from '$lib/actions/analytics';
+    import { Modal } from '$lib/components';
+    import { Dependencies } from '$lib/constants';
+    import { Button, FormList, InputText } from '$lib/elements/forms';
+    import { addNotification } from '$lib/stores/notifications';
+    import type { Attributes } from '../store';
+    import { options, type Option } from './store';
+
+    export let showEdit = false;
+    export let selectedAttribute: Attributes;
+
+    const databaseId = $page.params.database;
+    const collectionId = $page.params.collection;
+
+    let error: string;
+    let currentAttr: Attributes;
+
+    $: option = options.find((option) => {
+        if (selectedAttribute) {
+            if ('format' in selectedAttribute && selectedAttribute.format) {
+                return option?.format === selectedAttribute?.format;
+            } else {
+                return option?.type === selectedAttribute?.type;
+            }
+        }
+    }) as Option;
+
+    async function submit() {
+        try {
+            await option.update(databaseId, collectionId, selectedAttribute);
+            await Promise.allSettled([invalidate(Dependencies.COLLECTION)]);
+            if (!$page.url.pathname.includes('attributes')) {
+                await goto(
+                    `${base}/console/project-${$page.params.project}/databases/database-${databaseId}/collection-${collectionId}/attributes`
+                );
+            }
+            addNotification({
+                type: 'success',
+                message: `Attribute ${selectedAttribute.key} has been updated`
+            });
+            showEdit = false;
+            trackEvent(Submit.AttributeUpdate);
+        } catch (e) {
+            error = e.message;
+            trackError(e, Submit.AttributeUpdate);
+        }
+    }
+
+    $: if (showEdit) {
+        currentAttr ??= { ...selectedAttribute };
+        error = null;
+    }
+
+    function shouldBeDisabled(oldAttr: Attributes, newAttr: Attributes) {
+        return JSON.stringify(oldAttr) === JSON.stringify(newAttr);
+    }
```

Comment:
not sure I like this approach, it kinda feels like weird to use JSON strings for comparisons

Reply:
Do you have any suggestions on how to handle this best? :thinking: